sycl/include/sycl/ext/oneapi/matrix/query-types.hpp

```diff
@@ -42,5 +45,54 @@ struct combination {
 };
 
 } // namespace ext::oneapi::experimental::matrix
+
+namespace detail {
+template <typename T> constexpr const char *convertTypeToMatrixTypeString() {
```

Comment:
could we use traits to make the code cleaner?
```
template <typename T>
struct traits {
    static constexpr const char* value = "";
};

#define traits(T, String)                                    \
  template <> struct traits<T> {                              \
    static constexpr const char* value = String;                         \
  };

traits(int, "matrix_type::fp32")
```
https://godbolt.org/z/5q5MGdM9d

Reply:
I don't see how "struct" is better than function here. I'd make a single function and use "if constexpr (std::is_same_v<T, Type1>) {}" though.

Reply:
@aelovikov-intel looks good!

Reply:
> I don't see how "struct" is better than function here. I'd make a single function and use "if constexpr (std::is_same_v<T, Type1>) {}" though.

I did it as a single function initially in the early commits but didn't like these else if/else if/else if, so changed to the template specializations.

Reply:
I don't agree - https://godbolt.org/z/MYrdE64n7. That said, this is up to the "matrix" reviewers.

Reply:
I meant the pure code. With the macro it looks very clean, of course.